arch/arm/arm64/Makefile.uk

```diff
@@ -67,205 +67,236 @@ ARCHFLAGS-y += -mbranch-protection=$(BRANCH_PROTECTION)
 ISR_ARCHFLAGS-y += -mbranch-protection=$(BRANCH_PROTECTION)
 endif
 
-# GCC support -mcpu=native for arm64 from 6.0
 ifeq ($(CONFIG_MCPU_ARM64_NATIVE),y)
-$(call error_if_gcc_version_lt,6,0)
-ARCHFLAGS-$(call gcc_version_ge,6,0)     += -mcpu=native
-ISR_ARCHFLAGS-$(call gcc_version_ge,6,0) += -mcpu=native
+ARCHFLAGS     += -mcpu=native
+ISR_ARCHFLAGS += -mcpu=native
 endif
 
-# GCC support -mcpu=generic for arm64 from 4.8
 ifeq ($(CONFIG_MCPU_ARM64_GENERIC),y)
-$(call error_if_gcc_version_lt,4,8)
-ARCHFLAGS-$(call gcc_version_ge,4,8)     += -mcpu=generic -mtune=generic
-ISR_ARCHFLAGS-$(call gcc_version_ge,4,8) += -mcpu=generic -mtune=generic
+ARCHFLAGS     += -mcpu=generic -mtune=generic
+ISR_ARCHFLAGS += -mcpu=generic -mtune=generic
 endif
 
-# GCC support -mcpu=cortex-a53 for arm64 from 4.9
 ifeq ($(CONFIG_MCPU_ARM64_CORTEX_A53),y)
-$(call error_if_gcc_version_lt,4,9)
-ARCHFLAGS-$(call gcc_version_ge,4,9)     += -mcpu=cortex-a53 -mtune=cortex-a53
-ISR_ARCHFLAGS-$(call gcc_version_ge,4,9) += -mcpu=cortex-a53 -mtune=cortex-a53
+ARCHFLAGS     += -mcpu=cortex-a53 -mtune=cortex-a53
+ISR_ARCHFLAGS += -mcpu=cortex-a53 -mtune=cortex-a53
 endif
 
 # For erratum 835769
 ifeq ($(CONFIG_ARM64_ERRATUM_835769),y)
-$(call error_if_gcc_version_lt,4,9)
-ARCHFLAGS-$(call gcc_version_ge,4,9)     += -mfix-cortex-a53-835769
-ISR_ARCHFLAGS-$(call gcc_version_ge,4,9) += -mfix-cortex-a53-835769
+ARCHFLAGS     += -mfix-cortex-a53-835769
+ISR_ARCHFLAGS += -mfix-cortex-a53-835769
 endif
 
 # For erratum 843419
 ifeq ($(CONFIG_ARM64_ERRATUM_843419),y)
-$(call error_if_gcc_version_lt,4,9)
-ARCHFLAGS-$(call gcc_version_ge,4,9)     += -mfix-cortex-a53-843419
-ISR_ARCHFLAGS-$(call gcc_version_ge,4,9) += -mfix-cortex-a53-843419
+$(call error_if_gcc_version_lt,7,0)
```

Comment:
The `-mfix` for the `843419` errata is available only for `gcc`, `clang` does not offer it.
I added that check (and a few others like this one) in order for the build to fail if using `clang`.

I can do something more explicit like
```make
ifeq ($(call have_clang),y))
$(error Clang not supported with [...] enabled, use GCC >= 7.0)
endif
```

Reply:
Yes, that's way more clear. I would also avoid specifying the version in case we bump it later and forget to update the message, unless you can get obtain it via a variable. Otherwise saying something like "feature xyz only supported in GCC" would suffice.

Reply:
Yes, makes sense. I'll update the commits.